hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -196,8 +204,14 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
 
 return_type GenericSystem::read()
 {
-  // do loopback
-  for (size_t i = 0; i < joint_states_.size(); ++i) {
+  // apply offset to positions only
+  for (size_t j = 0; j < joint_states_[0].size(); ++j) {
```

Comment:
```suggestion
  for (size_t j = 0; j < joint_states_[POSITION_INTERFACE_INDEX].size(); ++j) {
```
helps readability a lot